### PR TITLE
Added minimum MSVC support

### PR DESCRIPTION
**Pull request by cole-clark**
Modified the minimum amount of code necessary to get this project to build on Windows through MSVC.

One known error that occurs at this point is that a python error is raised in both runtime and testing:

> ImportError: cannot import name '_usdTri' from 'usdpluginexamples.UsdTri' (D:\usd_projects\USDPluginExamples\20.11_windows_compatibility_install\lib\python\usdpluginexamples\UsdTri\__init__.py)
> Loading plugin 'usdImaging'.

Otherwise, however the other parts of this plugin appear to work as expected.

### CODE REVIEW

**Comment by rlei-weta** [comment]
Hey @colevfx - thanks for the submission, we really appreciate the work!  

It looks like one of the CI checks is failling, when you get a chance could you take a look to see what might be causing it?

File: README.md
Comment (rlei-weta) [inline README.md]: Cheers for adding this example build configuration snippet!

Do you think we could update the _source directory_ argument so that it is symmetrical to the above Linux/MacOS based snippet?  So, using the relative pathing:
```
..
```
Instead of the full absolute path:
```
"D:\usd_projects\USDPluginExamples\USDPluginExamplesMain"
```

File: README.md
Comment (rlei-weta) [inline README.md]: Thanks for adding this!

Since these variables is for the runtime, would "Windows" be a more appropriate platform specification than "MSVC"?

File: README.md
Comment (rlei-weta) [inline README.md:93]: @colevfx - would you mind also separating the `LD_LIBRARY_PATH` var into a _Linux_ specific table (given that it does not generally apply to other platforms) - thanks!

File: cmake/USDPluginTools.cmake
Comment (rlei-weta) [inline cmake/USDPluginTools.cmake:26]: This is a _much_ more readable and maintainable way of setting these paths.  Nice!

File: cmake/USDPluginTools.cmake
Comment (rlei-weta) [inline cmake/USDPluginTools.cmake:301]: @colevfx - I suspect the linux tests are failing because the `PYTHONPATH=...` and `PXR_PLUGINPATH_NAME=...` specification were not delimited by a semicolon `;`...

File: cmake/USDPluginTools.cmake
Comment (rlei-weta) [inline cmake/USDPluginTools.cmake]: Thanks for adding this, it exposes an existing issue on this project.  We should update the standard to C++14 for all platforms, given that is the requirement in the latest versions of USD.

Somewhere in this file there should be a:
```cmake
    target_compile_features(${TARGET_NAME}
        PRIVATE
            cxx_std_11
    )
```

Could you please update that to `cxx_std_14` (and remove this MSVC-specific setting) - cheers

File: cmake/USDPluginTools.cmake
Comment (rlei-weta) [inline cmake/USDPluginTools.cmake]: Cheers for adding the comment / explanation - it'll definitely help future maintainers.

Could you move this MSVC-specific definition addition into the `_usd_target_properties` implementation, and could you write the logic to pass these additional definitions into the `target_compile_definitions` call?  Thanks!

File: README.md
Comment (cole-clark) [reply README.md]: I think you are right! "Windows 10" is the platform, and MSVC is just the compiler after all

File: cmake/USDPluginTools.cmake
Comment (cole-clark) [reply cmake/USDPluginTools.cmake:301]: Ah perfect! I'll give that a try! Thanks for spotting that!

**Comment by cole-clark** [comment]
@rlei-weta Thank you very much for the PR notes! I'm going to hit these as soon as I can, but it might take me a day or two--apologies for the delay 😄

**Comment by rlei-weta** [comment]
@colevfx - cheers, take your time :-)

Oh - before I forget, when you make the update could you re-author the commits such that they contain a [DCO](https://en.wikipedia.org/wiki/Developer_Certificate_of_Origin).  You'll need to pass in the `--signoff` flag for a `git commit` operation. (And force-push once you've updated the commits)

It's totally on us that we haven't yet added official contributing guidelines - something we'll be looking at soon.  Thanks again!!

File: cmake/USDPluginTools.cmake
Comment (rlei-weta) [inline cmake/USDPluginTools.cmake]: @colevfx  - thanks for fixing up the missing semicolon.  

I was wondering... if there was a cleaner way to express the multiple environment variables we're setting (extending on your great idea around joining token lists).

What would you think of something to this effect:
```cmake
# Create a list of python path values.
set(TEST_PYTHON_PATHS 
    "${PROJECT_BINARY_DIR}/${CMAKE_INSTALL_LIBDIR}/python" 
    "${USD_ROOT}/${CMAKE_INSTALL_LIBDIR}/python" 
    "$ENV{PYTHONPATH}"
)

# Join python path values into a single string.
if (MSVC)
    list(JOIN TEST_PYTHON_PATHS ";" TEST_PYTHON_PATHS)
else()
    list(JOIN TEST_PYTHON_PATHS ":" TEST_PYTHON_PATHS)
endif()

# Join multiple environment variable strings into a single one.
list(JOIN ENV_PATHS ";"
    "PYTHONPATH=${TEST_PYTHON_PATHS}" 
    "PXR_PLUGIN_PATH=${_TEST_PXR_PLUGIN_PATH}"
)

set_tests_properties(${TEST_TARGET}
        PROPERTIES
            ENVIRONMENT
                ${ENV_PATHS}
)
```

This way we can de-couple the _joining of values per-env-var_ from the _joining of different env-vars_, which might clean up the logic a tiny bit.

File: cmake/USDPluginTools.cmake
Comment (rlei-weta) [inline cmake/USDPluginTools.cmake]: Ohh I haven't seen the two backslashes before.  Is that required? (Could you explain what it is for?)

**Review by rlei-weta — CHANGES_REQUESTED** [review]
Hey @colevfx thanks for pushing the updates - and looks like the CI pipeline passed too which is great.  I just had two more notes for you.  Once those are addressed I think this PR is ready to merge :)

File: cmake/USDPluginTools.cmake
Comment (cole-clark) [reply cmake/USDPluginTools.cmake]: Sure! So last night I bumbled through getting the Linux tests to pass a few times (sorry if there were a few extra emails!), and the double-escape sequence is here because of the output in:

This functioning line:
https://github.com/wetadigital/USDPluginExamples/pull/23/checks?check_run_id=1811841093#step:7:18

And this non-functioning line with triple colons:
https://github.com/wetadigital/USDPluginExamples/runs/1811709239?check_suite_focus=true#step:7:18

I'm a little fuzzy on the specifics, but my working theory is that CMake treats `;` as a list separator when lexing the code, and you first need to escape that functionality with a `\`, and then you need to escape the semicolon character with another `\`. This seems to match random SO searches and the language reference here:

https://cmake.org/cmake/help/latest/manual/cmake-language.7.html#escape-sequences

On MSVC, the common wisdom seems to be that you actually HAVE to double escape a semicolon in a string which is why I set it as the path separator, and my code wouldn't build when I didn't do it basically.

I did not test a single-escape here admittedly, but this seemed to be the safest choice for Linux and Windows 😄

File: cmake/USDPluginTools.cmake
Comment (cole-clark) [reply cmake/USDPluginTools.cmake]: I actually played with this idea myself yesterday!

I decided _not_ to do it then because it seemed to me that if we scaled this pattern to more variables, we'd be adding a set command, a list command, and modifying the final join statement for every variable we are adding. It seemed like an anti-pattern to me to have to do three modifications instead of one.

The advantage to what you wrote is that it's more semantic and easy to read, and truthfully I don't think it's likely we'd be adding more env vars here. But the disadvantage is that if we did adding more env vars would be slightly harder.

If you think that trade off is worth it, I'm happy to revisit this tonight and make this work 😄

Comment (rlei-weta) [reply cmake/USDPluginTools.cmake]: Thanks @colevfx, we definitely want to strive for more readability here (at the cost of more operations) - so if you could revisit this we'd appreciate it!  This might also fix some of the trailing colons in the test runs.

If we end up with more environment variables, we could try combining initial `set` and `list` into a single utility function `_set_env_var_paths` which should clean that up just a bit further.

File: cmake/USDPluginTools.cmake
Comment (rlei-weta) [reply cmake/USDPluginTools.cmake]: Thanks for the research & explanation, that makes sense! :-)

File: cmake/USDPluginTools.cmake
Comment (cole-clark) [reply cmake/USDPluginTools.cmake]: Gave this a go tonight, but had some serious problems trying to get it working in the manner you originally suggested. So what I tried doing is building out the path more semantically, but keeping the original overall command structure that I had before. I'm hoping that this is in the spirit of what you wanted, making it more explicit what's happening and why, even if it's not in the form you requested exactly. I'd love to know what you think😄 

TL;DR on why the other setup it wasn't working great was that the when you list(JOIN ...) the multiple environment path strings we were building, it wasn't just gluing the two environment variable paths, but also all of the sub-elements of each list which doesn't work. If need be I can share further details, and below is my non-functioning version of the code that seems okay on Windows, but won't pass Linux tests. I still prefer what I actually put into the PR though at this moment.

>     # Set-up runtime environment variables for the test.
>     # The paths refer to the build tree (which mirrors the final installation).
>     set(TEST_PYTHON_PATHS
>         "${PROJECT_BINARY_DIR}/${CMAKE_INSTALL_LIBDIR}/python" 
>         "${USD_ROOT}/${CMAKE_INSTALL_LIBDIR}/python" 
>         "$ENV{PYTHONPATH}")
>     
>     if (MSVC)
>         list(JOIN TEST_PYTHON_PATHS ";" TEST_PYTHON_PATHS)
>     else()
>         list(JOIN TEST_PYTHON_PATHS ":" TEST_PYTHON_PATHS)
>     endif()
> 
>     # Join multiple environment variable strings
>     set(ENV_PATHS
>         "PYTHONPATH=${TEST_PYTHON_PATHS}"
>         "PXR_PLUGINPATH_NAME=${_TEST_PXR_PLUGIN_PATH}")
>     list(JOIN ENV_PATHS "\\;" ENV_PATHS) 
> 
>     set_tests_properties(${TEST_TARGET}
>         PROPERTIES
>             ENVIRONMENT
>             "${ENV_PATHS}")

Comment (rlei-weta) [reply cmake/USDPluginTools.cmake]: Hey @colevfx - hehe that's what I get for proposing untested code.  

Thanks for finding a different and working solution for this, breaking it down into logical blocks, and keeping it readable :-)  It looks great!

**Review by rlei-weta — APPROVED** [review]

**Comment by rlei-weta** [comment]
Merging away, thanks again for the awesome work @colevfx!

**Pull request merged**
